### ISSUE TITLE
Scaling issues on the remarkable 2

### ISSUE THREAD

**Issue by iHnR** [koreader/koreader#10617]
* KOReader version: 2023.05.1
* Device: reMarkable2

#### Issue
This issue has two parts that I believe are related.  
1. First the auto DPI doesn't seem to work correctly on the remarkable. I have to manually set it to the native 226 to get a sensible UI scale (on auto it's really big). This is strange as I can find the right config in `frontend/device/remarkable/device.lua`.
2. After manually setting the DPI to 226, the font size in the OPDS and Cloud menus is really tiny (i guess all menus that use this ui class but I'm not sure what it's called). I tried reproducing this under linux and android but it seems it works fine there.

#### Steps to reproduce
1. Set DPI to 226.
2. Open the OPDS catalog and observe the really tiny font size.

**Comment by NiLuJe** [comment]
Define "really big" (i.e., post screenshot).

I have a vague inkling it might just be working as intended ;).

(Because, yes, IIRC, on 10" panels, the defaults feel sliiightly larger than usual from my own experience on the Elipsa).

**Comment by hius07** [comment]
OPDS and Cloud menus font size can be set in the upper menu Filebrowser (the first icon) - Settings - Classic mode settings - Item font size.

**Comment by iHnR** [comment]
> OPDS and Cloud menus font size can be set in the upper menu Filebrowser (the first icon) - Settings - Classic mode settings - Item font size.

Thanks. That fixes it. Not sure how it got set to such a small font in the first place. Finding the right setting in Koreader can be hard, but that's not so much a bug but a ux improvement.

**Comment by hius07** [comment]
> Not sure how it got set to such a small font in the first place.

If not set manually, it is calculated based on the "items per page" value (14 items per page by default).
https://github.com/koreader/koreader/blob/4ae25794a81c12af99fe9ca6c33edcd298f661eb/frontend/ui/widget/menu.lua#L1395-L1399

**Comment by iHnR** [comment]
> Define "really big" (i.e., post screenshots).
> 
> I have a vague inkling it might just be working as intended ;).
> 
> (Because, yes, IIRC, on 10" panels, the defaults feel sliiightly larger than usual from my own experience on the Elipsa).

Below is a comparison between auto dpi and manually setting to 226. I think it's substantial enough. Even if you think it's small, it's strange that it doesn't default to the native dpi. The code does specify the right dpi, so I guess that would be the intended behaviour.

![IMG_20230627_091118_1](https://github.com/koreader/koreader/assets/62661051/660a4b3a-a20c-460d-9ca5-61051155d6cc)
![IMG_20230627_090958](https://github.com/koreader/koreader/assets/62661051/dbd0c10d-6599-4fb3-a1ae-5917adcbcc8c)

**Comment by iHnR** [comment]
> > Not sure how it got set to such a small font in the first place.
> 
> If not set manually, it is calculated based on the "items per page" value (14 items per page by default).
> 
> https://github.com/koreader/koreader/blob/4ae25794a81c12af99fe9ca6c33edcd298f661eb/frontend/ui/widget/menu.lua#L1395-L1399

Yah my guess is that I changed a something a while ago and forgot about it. I edited out this part of the issue. Still keeping open since I think the dpi defaulting to the wrong value is a bug.

**Comment by NiLuJe** [comment]
Yup, that roughly looks like what I'm seeing no an Elipsa, FWIW.

**Comment by iHnR** [comment]
Thanks for the input @NiLuJe! It's not the biggest issue (I used the app for quite some time before realizing I could change this) but I do think it makes a substantial difference in UX and it would be nice to have out of the box. I tried to see if I can do a quick fix myself, but I haven't been able to wrap my head around the code base so far.

**Comment by poire-z** [comment]
I think most of our sizes (width/height, font size) go from some fixed abstract value (ie: 20) we use in the code into these `scaleBySize(the_abstract_value)` function to get the final real size in pixels:
https://github.com/koreader/koreader-base/blob/d3744561601e5c97219899b02a608cd94c2b02e0/ffi/framebuffer.lua#L395-L418

I/we can't really advocate for the existing current logic in there, it has possibly been like that since the beginning of KOReader.
We haven't changed it because it's rather ok - and the day we change it, lots of existing users may complain because the way it looks (with the default) has changed, and we are probably all used and fine with what we have. A slight change in size, menu or icons a bit bigger or smaller, and we won't feel at home :) So quite too risky to go at changeing that imho.
